### PR TITLE
 pkg/process: Add process-agent related packages 

### PR DESCRIPTION
**Pull request by sfluor**
### What does this PR do?

This adds all the packages from the `datadog-process-agent` repo.

This is the list of packages that got moved from the process-agent repo into `pkg/process`

```
github.com/DataDog/datadog-process-agent/checks
github.com/DataDog/datadog-process-agent/config
github.com/DataDog/datadog-process-agent/model
github.com/DataDog/datadog-process-agent/net
github.com/DataDog/datadog-process-agent/statsd
github.com/DataDog/datadog-process-agent/util
```

Along with those `github.com/DataDog/datadog-process-agent/ebpf` was moved into `pkg/ebpf`

### Motivation

Merging the `process-agent` code into the core agent repo

### CODE REVIEW

**Comment by sunhay** [comment]
Per the description, can we actually leave out cmd/agent and cmd/network-tracer from being put in /pkg/process?

Not sure if it was a typo since I don’t actually see the files in the changeset

**Comment by sfluor** [comment]
This was a typo sorry I will remove them from the description

**Comment by leeavital** [comment]
Some things are missing from the old repo:
 
- It looks like you checked in the `.o` files. They should be built as part of CI.
- there are some rake tasks (namely for building the tracer and agent, and compiling the BPF)  that haven't been ported over
- c-formatting is gone
- The old rake task `ebpf:test` isn't being run

**Comment by sfluor** [comment]
Good catch for the `.o` files for the tooling this will be in a follow up PR (we will probably want to get rid off the Rakefile and use invoke tasks too)

**Review by leeavital — DISMISSED** [review]

**Review by leeavital — DISMISSED** [review]

File: pkg/process/config/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Agent Configuration
```
Comment (sunhay) [inline pkg/process/config/README.md]: This file can be deleted

**Review by sunhay — DISMISSED** [review]
One change but LGTM

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -41,6 +41,7 @@
 /pkg/logs/                              @DataDog/logs-intake
 /pkg/metadata/ecs/                      @DataDog/burrito
 /pkg/metadata/kubernetes/               @DataDog/burrito
+/pkg/process/                           @DataDog/burrito
```
Comment (mfpierre) [inline .github/CODEOWNERS]: should we add `/pkg/ebpf` as well?

File: pkg/dogstatsd/listeners/uds_common_test.go [hunk 3]
```diff
@@ -3,7 +3,7 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-2019 Datadog, Inc.
 
-// +build !windows
+// +build linux
```
Comment (mfpierre) [inline pkg/dogstatsd/listeners/uds_common_test.go]: why do we need to change this?

File: .github/CODEOWNERS (hunk 2 above)
Comment (sfluor) [reply .github/CODEOWNERS]: 👍

File: pkg/dogstatsd/listeners/uds_common_test.go (hunk 3 above)
Comment (sfluor) [reply pkg/dogstatsd/listeners/uds_common_test.go]: Woops sorry wrong file will change, thanks for catching this !

**Review by mfpierre — APPROVED** [review]

**Review by sunhay — APPROVED** [review]

**Review by olivielpeau — APPROVED** [review]
Excited to see this merge happen! LGTM!

(I only reviewed the files outside of `pkg/process/` and `/pkg/ebpf`)

**Pull request merged**